### PR TITLE
Batched feature extraction for s3prl

### PR DESCRIPTION
**Pull request by desh2608**
This PR enables batch extraction of S3PRL features (similar to kaldifeat).

### CODE REVIEW

**Comment by desh2608** [comment]
@DongjiGao can you take a look?

File: lhotse/features/ssl.py [hunk 1]
```diff
@@ -68,35 +68,102 @@ def __init__(self, config: Optional[Any] = None):
...
+            ]
+        else:
+            # The user passed a single array/tensor of shape (num_samples,)
+            samples = [
+                torch.from_numpy(samples)
+                if isinstance(samples, np.ndarray)
+                else samples
+            ]
+
+        samples = [s.to(self.config.device) for s in samples]
```
[the remark this answers is not in the document]
Comment (desh2608) [reply lhotse/features/ssl.py:139]: I think it internally pads the list of tensors to the same length using `torch.rnn.utils.pad_sequence`.

**Comment by DongjiGao** [comment]
> @DongjiGao can you take a look?

LGTM.

**Comment by pzelasko** [comment]
Thanks!

**Pull request merged**